### PR TITLE
Assignment2 - new StockFactory directory

### PR DESCRIPTION
**Pull request by MikeBrooksNH**
All the code for the Stock Factory project has been added to the StockFactory subdirectory off the root folder.

### CODE REVIEW

**Comment by smarks** [comment]
delete the .idea directory and update this PR.

**Comment by MikeBrooksNH** [comment]
I removed the .idea directory from the root folder of this repo, but the root folder isn't where the assignment 2 project is located. The assignment 2 specific project is in the StockFactory folder.

The root folder contained the "x Collector" project - I'm not sure why the .idea folder was readded to that directory (the .gitignore file had *.idea in it).

**Comment by MikeBrooksNH** [comment]
This should be complete.

The .gitignore to  not include .iml files and removed the .iml files

**Comment by MikeBrooksNH** [comment]
@smarks am I good to merge this?

File: StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+package edu.mbrooks.advancedjava.StockFactory;
+import edu.mbrooks.advancedjava.StockQuote.*;
+import edu.mbrooks.advancedjava.StockFactory.*;
+import edu.mbrooks.advancedjava.StockService.*;
+
+/**
+ *  The StockTrader class implements the SockService interface
+ */
+class StockTrader implements StockService {
```
Comment (smarks) [inline StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java]: this is not quite the right approach, but we talked about this in chat, so I think you are OK now.

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (smarks) [inline .idea/vcs.xml]: don't include this file.

File: StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java (hunk 1 above)
Comment (MikeBrooksNH) [reply StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java]: Yup, renamed the class in assignment 3

**Pull request merged**